api/features/tests/test_views.py

```diff
@@ -314,6 +314,29 @@ def test_audit_log_created_when_feature_updated(self):
             == 1
         )
 
+    def test_audit_logs_created_when_feature_deleted(self):
+        # Given
+        feature = Feature.objects.create(name="test feature", project=self.project)
+        feature_states_ids = feature.feature_states.values_list("id", flat=True)
+        # When
+        self.client.delete(
+            self.project_feature_detail_url % (self.project.id, feature.id)
+        )
+        # Then
+        # Audit log exists for the feature
+        assert AuditLog.objects.get(
+            related_object_type=RelatedObjectType.FEATURE.name,
+            related_object_id=feature.id,
+        )
+        # and audit logs exists for all feature states for that feature
+        assert (
+            AuditLog.objects.filter(
+                related_object_type=RelatedObjectType.FEATURE.name,
```

Comment:
I think this test is only passing because we only have a single environment and hence a single feature state. If you created another environment above, I think this test would fail because this line should be: 

```suggestion
                related_object_type=RelatedObjectType.FEATURE_STATE.name,
```

Reply:
Yep, missed it.